ote_cli/ote_cli/tools/demo.py

```diff
@@ -0,0 +1,200 @@
+        help="Source of input data: images folder, image, webcam and video.",
+    )
+    parser.add_argument(
+        "--load-weights",
+        required=True,
+        help="Load only weights from previously saved checkpoint",
+    )
+    parser.add_argument(
+        "--labels", nargs="+", required=True, help="A space-separated labels list."
+    )
+    parser.add_argument(
+        "--fit-to-size",
+        nargs=2,
+        type=int,
+        help="Width and Height space-separated values. "
+        "Fits displayed images to window with specified Width and Height. "
+        "This options applies to result visualisation only.",
+    )
+    parser.add_argument(
+        "--loop", action="store_true", help="Enable reading the input in a loop."
+    )
+    parser.add_argument(
+        "--delay", type=int, default=3, help="Frame visualization time in ms."
+    )
+    parser.add_argument(
+        "--display-perf",
+        action="store_true",
+        help="This option enables writing performance metrics on displayed frame. "
+        "These metrics take into account not only model inference time, but also "
+        "frame reading, pre-processing and post-processing.",
+    )
+
+    add_hyper_parameters_sub_parser(parser, hyper_parameters, modes=("INFERENCE",))
+
+    return parser.parse_args(), template, hyper_parameters
+
+
+def get_predictions(task, frame):
+    """
+    Returns list of predictions made by task on frame and time spent on doing prediction.
+    """
+
+    empty_annotation = AnnotationSceneEntity(
+        annotations=[], kind=AnnotationSceneKind.PREDICTION
+    )
+
+    item = DatasetItemEntity(
+        media=Image(frame),
+        annotation_scene=empty_annotation,
+    )
+
+    dataset = DatasetEntity(items=[item])
+
+    start_time = time.perf_counter()
+    predicted_validation_dataset = task.infer(
+        dataset,
+        InferenceParameters(is_evaluation=True),
+    )
+    elapsed_time = time.perf_counter() - start_time
+    item = predicted_validation_dataset[0]
+    return item.get_annotations(), elapsed_time
+
+
+def main():
+    """
+    Main function that is used for model demonstration.
+    """
+
+    # Dynamically create an argument parser based on override parameters.
+    args, template, hyper_parameters = parse_args()
+    # Get new values from user's input.
+    updated_hyper_parameters = gen_params_dict_from_args(args)
+    # Override overridden parameters by user's values.
+    override_parameters(updated_hyper_parameters, hyper_parameters)
+
+    hyper_parameters = create(hyper_parameters)
+
+    labels = [
+        LabelEntity(l, template.task_type, id=ID(i)) for i, l in enumerate(args.labels)
+    ]
+
+    # Get classes for Task, ConfigurableParameters and Dataset.
+    task_class = get_impl_class(template.entrypoints.base)
+    environment = TaskEnvironment(
+        model=None,
+        hyper_parameters=hyper_parameters,
+        label_schema=LabelSchemaEntity.from_labels(labels),
```

Comment:
This wouldn't work for multilabel: classification makes a decision which config to used based in the label schema. To add multilabel classification this step should be customized. The same is true for hierarchical classification.

Reply:
Will work on this issue and Reading/Writing labels from/to snapshot in next PR(s)

Reply:
@druzhkov-paul